applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -192,6 +236,77 @@ static int handle_at_sleep(const char *at_cmd, enum shutdown_modes *mode)
 	return ret;
 }
 
+static int handle_at_slmuart(const char *at_cmd, u32_t *baudrate)
+{
+	int ret = -EINVAL;
+	enum at_cmd_type type;
+
+	ret = at_parser_params_from_str(at_cmd, NULL, &at_param_list);
+	if (ret < 0) {
+		LOG_ERR("Failed to parse AT command %d", ret);
+		return -EINVAL;
+	}
+
+	type = at_parser_cmd_type_get(at_cmd);
+	if (type == AT_CMD_TYPE_SET_COMMAND) {
+		if (at_params_valid_count_get(&at_param_list) > 1) {
+			ret = at_params_int_get(&at_param_list, 1,
+					baudrate);
+			if (ret < 0) {
+				LOG_ERR("AT parameter error");
+				return -EINVAL;
+			}
+		}
+		if (*baudrate == 1200) {
```

Comment:
IMO this construct is a bit bulky, how about something like:
```
if (*baudrate == 1200 || *baudrate == 2400 ||
     ...) {
```

Reply:
Yes it is bulky. The repeated || in if statement also looks bulky , so I changed the repeated statement to switch cases:

		switch (*baudrate) {
		case 1200:
		case 2400:
		case 4800:
		case 9600:
		case 14400:
		case 19200:
		case 38400:
		case 57600:
		case 115200:
		case 230400:
		case 460800:
		case 921600:
		case 1000000:
			ret = 0;
			break;
		default:
			LOG_ERR("Invalid uart baud rate provided.");
			return -EINVAL;
		}

How about above?